### PR TITLE
Remove zone support

### PR DESCRIPTION
**Pull request by Lagovas**
Here is removed acra-addzone tool, all zonemode related things from code, tests, benchmarks, docker files. 
Some of interfaces updated to support extra data `additional context` to be able to use AD (additional data) in AEAD (authenticated encryption with additional data) encryption. It is not used for now but left args for future extensions if we will want. But it is discussible do we need to leave it now...

It's draft because there is remaining changes in our `./examples` and `acrawriter` wrappers. But main removings already made.

P.S. I tried to split changes into separate commits due to a lot of changes. 

## Checklist

- [ ] Change is covered by automated tests
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation in the [Acra documentation] site or has PR on [documentation repository] 
  with new changes
- [ ] CHANGELOG.md is updated (in case of notable or breaking changes)
- [ ] CHANGELOG_DEV.md is updated
- [ ] Benchmark results are attached (if applicable)
- [ ] [Example projects and code samples] are up-to-date (in case of API changes) 

[coding guidelines]: https://golang.org/doc/effective_go
[Example projects and code samples]: https://github.com/cossacklabs/acra-engineering-demo
[Acra documentation]: https://docs.cossacklabs.com/
[documentation repository]: https://github.com/cossacklabs/product-docs

### CODE REVIEW

File: benchmarks/docker-compose.pgsql-ssl-server-ssl.yml [hunk 1]
```diff
@@ -40,7 +39,6 @@ services:
         build:
             context: ../
             dockerfile: docker/postgresql-ssl.dockerfile
-        network_mode: host
```
Comment (Zhaars) [inline benchmarks/docker-compose.pgsql-ssl-server-ssl.yml:43]: What is the reason for this change? As I know default `network_mode` in compose if `bridge` means that containers will be placed in the compose bridge network and they will not be accessible from the host.  So this removing changes the compose behavior - I am just trying to understand whether it isn`t breaking anything.

Comment (Lagovas) [reply benchmarks/docker-compose.pgsql-ssl-server-ssl.yml:43]: my docker-compose:
```
$ docker-compose -v
docker-compose version 1.29.2
```
tells me: `docker.errors.InvalidArgument: "host" network_mode is incompatible with port_bindings`
As I think, it is not valid anymore if you declare ports (that we do in this file). I tested locally that ports are accessible from the host and they are.

**Comment by Lagovas** [comment]
We decided to merge this PR in next release after deprecating Zones.
Additionaly, we should re-check our eng demos with these changes due to renaming several files in examples. Just a reminder for me.
